### DISCUSSION TITLE
Propagating Custom Headers in Responses

### DISCUSSION THREAD

**Discussion by aakp10** [istio/istio#45505] [General]
I've an application that adds some custom header fields in its HTTP responses for all its endpoints. However, it seems like when I access this service from outside the cluster, Istio doesn't send out the custom header field that the application had sent with its response.

I'd have expected that if any response is intercepted by the envoy proxy, its headers should be propagated as is by the proxy.

Is this an expected behavior?

**Comment by sovitacharya** [comment]
Can you attach the application here , which is sending some headers into your payload .

**Reply by sovitacharya** [reply]
I did using a go application that is adding a custom header and mapped the APPLICATION to authorization policy provider , so as to work it out, also check if you have configured namespaces properly(cuz I too had the same issue because of namespace mismatch).

**Reply by aakp10** [reply]
@sovitacharya thanks for your suggestion; it worked.
